container-search/src/main/java/com/yahoo/search/handler/SearchHandler.java

```diff
@@ -303,15 +292,12 @@ private HttpSearchResponse handleBody(HttpRequest request) throws JSONException
         if (request.getMethod() == com.yahoo.jdisc.http.HttpRequest.Method.POST && request.getHeader(com.yahoo.jdisc.http.HttpHeaders.Names.CONTENT_TYPE).equals(JSON_CONTENT_TYPE)) {
             Inspector inspector = null;
             try {
-                InputStream inputStream = request.getData();
-                ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
-                byte[] buffer = new byte[0xFFFF];
-                for (int len = inputStream.read(buffer); len != -1; len = inputStream.read(buffer)) {
-                    outputStream.write(buffer, 0, len);
-                }
-                inspector = SlimeUtils.jsonToSlime(outputStream.toByteArray()).get();
-
-            } catch (IOException e) { e.printStackTrace();
+                byte[] byteArray = IOUtils.readBytes(request.getData(), 1 << 20);
+                inspector = SlimeUtils.jsonToSlime(byteArray).get();
+
+            } catch (IOException e) {
+                e.printStackTrace();
```

Comment:
Exception messages are logged where they are caught, so you either:

1. throw an exception here, and take care to include the cause, i.e.g, `throw new RuntimeException("Some message", e);`, or
2. catch the exception and log it, using the `Logger log`. 

Which of the two depends on where you can deal with the exception. In this case, there isn't much you can do, and the result should be a 400 or a 500, depending on whether the input was wrong (jsonToSlime threw) or if the internal request object encountered an exception (readBytes threw). 

I guess it was a good idea to throw a separate exception for the JSON exception, so perhaps you should bring that back :)

Reply:
Another thing on logging: Never log on errors which are caused by the query. (Doing that makes the log unusable and worst case allows clients to DOS the system)